src/Time/Formatting.hs

```diff
@@ -147,7 +161,7 @@ instance ( KnownRatName unit
                     nextUnit = newUnit - flooredNewUnit
                 in if nextUnit == 0
                    then show newUnit
-                   else timeStr ++ seriesF @units @unit nextUnit
+                   else timeStr ++ seriesF @(nextUnit:units) @unit nextUnit
```

Comment:
Strange that you didn't saw warning for this... But personally I think that when working with types in mixed mode it's better to specify promotion explicitly, like `nextUnit ': units`.